udao_spark/optimizer/base_optimizer.py

```diff
@@ -523,29 +577,52 @@ def sample_theta_x(
         n_samples: int,
         theta_type: ThetaType,
         seed: Optional[int],
+        selected_features: Optional[Dict[str, List[str]]],
         normalize: bool = True,
         mode: str = "random",
     ) -> np.ndarray:
         if seed is not None:
             np.random.seed(seed)
-        if mode == "random":
-            samples = np.random.randint(
-                low=self.theta_minmax[theta_type][0],
-                high=self.theta_minmax[theta_type][1],
-                size=(n_samples, len(self.theta_minmax[theta_type][0])),
-            )
-        elif mode == "lhs":
-            # a trivial implementation of Latin Hypercube Sampling when
-            # all parameters(knobs) are integers
-            sampler = qmc.LatinHypercube(
-                d=len(self.theta_minmax[theta_type][0]), seed=seed
+
+        if selected_features is not None:
+            theta_params = self.theta_names[theta_type]
+            selected_ids = []
+            for c in selected_features[theta_type]:
+                if c not in theta_params:
+                    raise ValueError(f"{c} is not in {theta_params}")
+                selected_ids.append(theta_params.index(c))
+        else:
+            selected_ids = list(range(len(self.theta_names[theta_type])))
+
+        theta_min = self.theta_minmax[theta_type][0][selected_ids]
+        theta_max = self.theta_minmax[theta_type][1][selected_ids]
+        n_params = len(selected_ids)
+
+        samples = np.repeat(self.theta_default[theta_type], n_samples, axis=0)
+        if n_params == 0:
+            logger.warning(
+                f"no selected features for {theta_type}, force to use default"
             )
-            samples_normed = sampler.random(n_samples)
-            low = self.theta_minmax[theta_type][0]
-            high = self.theta_minmax[theta_type][1] + 1
-            samples = np.floor(samples_normed * (high - low) + low).astype(int)
         else:
-            raise ValueError(f"mode {mode} is not supported")
+            if mode == "random":
+                samples_selected = np.random.randint(
+                    low=theta_min,
+                    # fixme: high is exclusive, should be theta_max + 1
```

Comment:
needs to be fixed in a long run - the theta_max is currently excluded in the random sample.